### PR TITLE
Remove `tracing-opentelemetry` dependency from `tracing-grpc` example

### PR DESCRIPTION
**Pull request by lalitb**
Fixes (partially) #1413 

## Changes

A minimalistic version of `tracing-grpc` example without `opentelemetry-jaeger` and `tracing-*` dependencies

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: examples/tracing-grpc/src/client.rs [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+use opentelemetry::{
+    trace::{SpanKind, TraceContextExt, Tracer},
+    Context, KeyValue,
+};
+
+fn init_tracer() {
+    global::set_text_map_propagator(TraceContextPropagator::new());
+    // Install stdout exporter pipeline to be able to retrieve the collected spans.
+    let provider = TracerProvider::builder()
+        .with_simple_exporter(SpanExporter::default())
```
Comment (TommyCpp) [inline examples/tracing-grpc/src/client.rs:16]: Since we have tokio here can we use batch span processor here?

Comment (lalitb) [reply examples/tracing-grpc/src/client.rs]: Good point. Updated to use the batch span processor with Tokio runtime.

**Review by shaun-cox — APPROVED** [review]

File: examples/tracing-grpc/src/client.rs [hunk 2]
```diff
@@ -24,45 +39,46 @@ pub mod hello_world {
...
+        Ok(_res) => "OK".to_string(),
+        Err(status) => {
+            // Access the status code
+            let status_code = status.code();
+            status_code.to_string()
+        }
+    };
+    cx.span().add_event(
+        "Got response!".to_string(),
+        vec![KeyValue::new("status", status)],
```
Comment (cijothomas) [inline examples/tracing-grpc/src/client.rs:72]: I think we should be showing the recommended approach of storing this to Span status itself, if failed.

**Review by cijothomas — APPROVED** [review]
Looks good!

left couple of nits, non-blocking!

**Pull request merged**